apps/hash-integration-worker/src/activities.ts

```diff
@@ -1,6 +1,120 @@
-export const createIntegrationActivities = () => ({
-  // eslint-disable-next-line @typescript-eslint/require-await,@typescript-eslint/no-unused-vars
-  async helloWorldActivity(params: {}): Promise<String> {
-    return "Hello, world!";
+import {
+  Attachment,
+  Comment,
+  Connection,
+  CustomView,
+  Cycle,
+  Document,
+  Issue,
+  IssueLabel,
+  LinearClient,
+  LinearDocument,
+  Organization,
+  Project,
+  ProjectMilestone,
+  Team,
+  User,
+} from "@linear/sdk";
+
+const readNodes = async <T>(connection: Connection<T>): Promise<T[]> => {
+  const nodes = connection.nodes;
+  while (connection.pageInfo.hasNextPage) {
+    // eslint-disable-next-line no-param-reassign
+    connection = await connection.fetchNext();
+    nodes.push(...connection.nodes);
+  }
+  return nodes;
+};
+
+export const createLinearIntegrationActivities = (createInfo: {
+  linearClient: LinearClient;
+}) => ({
+  async me(): Promise<User> {
+    return await createInfo.linearClient.viewer;
+  },
+
+  async organization(): Promise<Organization> {
+    return await createInfo.linearClient.organization;
+  },
+
+  async teams(): Promise<Team[]> {
+    return await createInfo.linearClient.teams().then(readNodes);
+  },
+
+  async issues(filter?: { teamId?: string }): Promise<Issue[]> {
+    const issuesQueryVariables: LinearDocument.IssuesQueryVariables = {
+      filter: {},
+    };
+    if (filter && filter.teamId) {
```

Comment:
[The optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)
```suggestion
    if (filter?.teamId) {
```